dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/BlockingTaskProcessor.java

```diff
@@ -0,0 +1,208 @@
+        this.taskInstance.setHost(NetUtils.getAddr(masterConfig.getListenPort()));
+        this.taskInstance.setState(ExecutionStatus.RUNNING_EXECUTION);
+        this.taskInstance.setStartTime(new Date());
+        this.processService.saveTaskInstance(taskInstance);
+        this.dependentParameters = taskInstance.getDependency();
+        this.blockingConditionResult = taskInstance.getBlockingCondition();
+    }
+
+    /**
+     *
+     * @param item the dependent item containing depTasks and status
+     * @return depend result for depend item. SUCCESS or FAILED
+     */
+    private DependResult getDependentResultForItem(DependentItem item) {
+        DependResult dependResult = DependResult.SUCCESS;
+        if (!completeTaskList.containsKey(item.getDepTasks())) {
+            logger.info("depend item: {} have not completed yet.", item.getDepTasks());
+            return dependResult;
+        }
+        // the actual status of task
+        ExecutionStatus executionStatus = completeTaskList.get(item.getDepTasks());
+        if (executionStatus != item.getStatus()) {
+            logger.info("depend item: {} expect status: {}, actual status: {}",item.getDepTasks(),
+                    item.getStatus(),executionStatus);
+            dependResult = DependResult.FAILED;
+        }
+        logger.info("dependent item complete {} {}, {}",
+                Constants.DEPENDENT_SPLIT,item.getStatus(),executionStatus);
+        return dependResult;
+    }
+
+    private void setConditionResult() {
+
+        List<TaskInstance> taskInstances = processService.findValidTaskListByProcessId(taskInstance.getProcessInstanceId());
+        for (TaskInstance task : taskInstances) {
+            completeTaskList.putIfAbsent(task.getName(), task.getState());
+        }
+
+        List<DependResult> modelResultList = new ArrayList<>();
+        for (DependentTaskModel dependentTaskModel : dependentParameters.getDependTaskList()) {
+            List<DependResult> itemDependResult = new ArrayList<>();
+            for (DependentItem item : dependentTaskModel.getDependItemList()) {
+                itemDependResult.add(getDependentResultForItem(item));
+            }
+            DependResult modelResult = DependentUtils.getDependResultForRelation(dependentTaskModel.getRelation(), itemDependResult);
+            modelResultList.add(modelResult);
+        }
+        conditionResult = DependentUtils.getDependResultForRelation(dependentParameters.getRelation(), modelResultList);
+        logger.info("the conditions task depend result : {}", conditionResult);
+    }
+
+    /**
+     * set task execution status and then update db
+     * for blocking task, determine blocking logic result in this function
+     */
+    private void endTask() {
+        ExecutionStatus status = ExecutionStatus.SUCCESS;
+        DependResult expected = "BlockingOnSuccess".equals(this.blockingConditionResult) ? DependResult.SUCCESS : DependResult.FAILED;
```

Comment:
Replace hardcoded 'BlockingOnSuccess' to variable.

Reply:
If replace hardcode to variable, the code fragment maybe look like this:

```java
String var = "BlockingOnSuccess";
DependResult expected = var.equals(this.blockingConditionResult) ? DependResult.SUCCESS : DependResult.FAILED;
```
It is confused to me why using variable is better?
The hardcode `BlockingOnSuccess ` is received from task definition json string. I use this value to consider user expected `DependResult `. I think using hardcode and varivale are  the same.